apps/storefront/src/pages/QuotesList/index.test.tsx

```diff
@@ -270,6 +271,307 @@ describe('when the user is a B2B customer', () => {
     expect(within(rowOfEveryHat).getByText('Expired')).toBeInTheDocument();
   });
 
+  describe('when the user enters a search term', () => {
+    it('displays the quotes that match the search term', async () => {
+      const manySocks = buildQuoteEdgeWith({ node: { quoteTitle: 'Many Socks' } });
+      const someTrousers = buildQuoteEdgeWith({ node: { quoteTitle: 'Some Trouser' } });
+
+      const allQuotes = buildQuotesListB2BWith({
+        data: { quotes: { totalCount: 2, edges: [manySocks, someTrousers] } },
+      });
+
+      const getQuotesList = vitest.fn<unknown[], QuotesListB2B>().mockReturnValue(allQuotes);
+
+      const getQuotesListQuerySpy = vitest.fn();
+
+      server.use(
+        graphql.query('GetQuotesList', ({ query }) => {
+          getQuotesListQuerySpy(query);
+
+          return HttpResponse.json(getQuotesList());
+        }),
+        graphql.query('GetShoppingListsCreatedByUser', () =>
+          HttpResponse.json(buildShoppingListsCreatedByUserWith('WHATEVER_VALUES')),
+        ),
+      );
+
+      renderWithProviders(<QuotesList />, { preloadedState });
+
+      const searchField = await screen.findByPlaceholderText('Search');
+      const table = await screen.findByRole('table');
+
+      expect(within(table).getByRole('row', { name: /Many Socks/ })).toBeInTheDocument();
+      expect(within(table).getByRole('row', { name: /Some Trouser/ })).toBeInTheDocument();
+
+      getQuotesList.mockReturnValue(
+        buildQuotesListB2BWith({
+          data: { quotes: { totalCount: 1, edges: [manySocks] } },
+        }),
+      );
+
+      await userEvent.type(searchField, 'Many Socks');
+
+      await waitFor(() =>
+        expect(within(table).queryByRole('row', { name: /Some Trouser/ })).not.toBeInTheDocument(),
+      );
+
+      expect(within(table).getByRole('row', { name: /Many Socks/ })).toBeInTheDocument();
+
+      expect(getQuotesListQuerySpy).toHaveBeenCalledWith(
```

Comment:
Do we know if mocks are reset after `mockReturnValue`? if not, what about using something like: 

```suggestion
      expect(getQuotesListQuerySpy).toHaveBeenCalledLastWith(
```